pkg/service/backup/restore_worker_tables.go

```diff
@@ -0,0 +1,158 @@
+package backup
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/scylladb/go-set/strset"
+	"github.com/scylladb/scylla-manager/v3/pkg/service/backup/backupspec"
+)
+
+type tablesWorker struct {
+	restoreWorkerTools
+
+	hosts   []restoreHost // Restore units created for currently restored location
```

Comment:
After separating `tablesWorker` and `indexWorker` there is no actual need to keep `hosts` as a field in `tablesWorker`, because we only pass it straight to the `indexWorker` right after its initialization.
This would also fit into the theme of separating those two structures.